File: ComInLanWindows/ComInLan/Model/CClient.cs
using ComInLan.Model.Base;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Text;
using System.Threading.Tasks;

namespace ComInLan.Model
{
	public class CClient : BaseModel, IClient
	{
		//--------------Created from packet------------------//
		public string Id { get; set;  }

		public string Name { get; set; }

		public int Port { get; set; }

		public CClient()
		{
			State = ClientState.None;
		}

		//--------------Created by app------------------//
		public IPAddress Address { get; set; }

		private string _passcode;
		public string Passcode
		{
			get
			{
				return _passcode;
			}
			set
			{
				_passcode = value;

				if (PasscodeCreated != null)
				{
					PasscodeCreated(this);
				}
			}
		}

		private ClientState _state;
		public ClientState State
		{
			get { return _state; }
			set
			{
				_state = value;

				if (StateChanged != null)
				{
					StateChanged(this);
				}
			}
		}

		public string Checksum { get; private set; }

		public long RefreshTime { get; private set; }

		//--------------Methods------------------//
		public void Refresh()
		{
			RefreshTime = GetCurrentUnixTimestamp();
		}

		public void CalculateChecksum()
		{
			Checksum = CalculateChecksum(Id + Name);
		}

		//--------------Events------------------//
		public event ClientPasscodeEventHandler PasscodeCreated;
		public event ClientStateEventHandler StateChanged;
		public event ClientDataEventHandler DataReceived;

		public void CallDataReceived(String dataJson)
		{
			if (DataReceived != null)
			{
				DataReceived(this, dataJson);
			}
		}
	}
}

File: ComInLanDroid/cominlan/src/main/java/com/onballgroup/cominlan/model/CClient.java
package com.onballgroup.cominlan.model;

import com.onballgroup.cominlan.model.Base.BaseModel;

import java.net.InetAddress;

/**
 * Created by Phong Le on 4/20/2016.
 */
public class CClient extends BaseModel implements IClient {
    private String _id;
    private String _name;
    private int _port;

    public CClient(){
        _state = ClientState.None;
    }

    @Override
    public String getId() {
        return _id;
    }

    public void setId(String id) {
        _id = id;
    }

    @Override
    public String getName() {
        return _name;
    }

    public void setName(String name) {
        _name = name;
    }

    @Override
    public int getPort() {
        return _port;
    }

    public void setPort(int port) {
        _port = port;
    }

    //--------------Created by app------------------//
    private InetAddress _address;
    private String _passcode;
    private ClientState _state;
    private String _checksum;
    private long _refreshTime;

    @Override
    public InetAddress getAddress() {
        return _address;
    }

    public void setAddress(InetAddress address) {
        _address = address;
    }

    @Override
    public String getPasscode() {
        return _passcode;
    }

    public void setPasscode(String passcode) {
        _passcode = passcode;

        if (_onClientListener != null)
        {
            _onClientListener.onPasscodeCreated(this);
        }
    }

    @Override
    public ClientState getState() {
        return _state;
    }

    public void setState(ClientState state)
    {
        _state = state;

        if (_onClientListener != null)
        {
            _onClientListener.onStateChanged(this);
        }
    }

    @Override
    public String getChecksum() {
        return _checksum;
    }

    @Override
    public long getRefreshTime() {
        return _refreshTime;
    }


    //--------------Methods------------------//
    public void calculateChecksum() {
        _checksum = calculateChecksum(_id + _name);
    }

    public void refresh() {
        _refreshTime = getCurrentUnixTimestamp();
    }

    //--------------Events------------------//
    private  OnClientListener _onClientListener;

    @Override
    public void setOnClientListener(OnClientListener listener) {
        _onClientListener = listener;
    }

    public void callDataReceived(String dataJson)
    {
        if (_onClientListener != null)
        {
            _onClientListener.onDataReceived(this);
        }
    }
}
